### ISSUE TITLE
Add support for proxy

### ISSUE THREAD

**Issue by ngosang** [FlareSolverr/FlareSolverr#26]
Add new parameters in the request:
* proxy Type, URL and credentials

Configure Chrome to use them

https://blog.scrapinghub.com/how-to-use-a-proxy-in-puppeteer

**Comment by ngosang** [comment]
@abeloin I see we already have some code to configure the proxy but it's not documented.

**Comment by abeloin** [comment]
Yes with something like this:
```
proxy:
{
  "url": "proxy.host",
  "username": "optional",
  "password": "only_with_username"
}
```
The current implementation naively expect http + basic authentication.

But from what I've been reading:

- http does work (https://github.com/puppeteer/puppeteer/issues/3253)
- https might work if we add `ignoreHTTPSErrors:true,` in puppeteer.launch (https://stackoverflow.com/questions/52368847/puppeteer-page-authenticate-https-proxy-not-work)

**Comment by ngosang** [comment]
Thank you! This is a low priority for me.
In the future we can configure FlareSolverrSharp with Jackett proxy so both use the same proxy and it's transparent to the user.

**Comment by DRSDavidSoft** [comment]
Another option that I'm interested in is to also support the system-wide proxy (e.g. Windows), or by reading the `$http_proxy` and `$HTTPS_PROXY` environment variables.

**Comment by johnRambo2k14** [comment]
Well I just made fork that runs the latest flaresolverr Linux  binaries in a openvpn / wireguard docker image. 

Not exactly a proxy but it helped me get around flaresolverr not being able to access certain sites to due to restrictions by my ISP.

Here's my repo: https://github.com/johnRambo2k14/docker-FlareSolverrVpn

**Comment by ngosang** [comment]
Implemented in [FlareSolverr v2.0.0](https://github.com/FlareSolverr/FlareSolverr/releases/tag/v2.0.0)
If you are using Jackett, version v0.18.1041 or higher is required to bypass Cloudflare.

**Comment by almico** [comment]
> Implemented in [FlareSolverr v2.0.0](https://github.com/FlareSolverr/FlareSolverr/releases/tag/v2.0.0)

@ngosang When you say "implemented", do you include support for "credentials", as you asked for in your first post?
